src/xo-app/settings/config/index.js

```diff
@@ -0,0 +1,138 @@
+import _ from 'intl'
+import ActionButton from 'action-button'
+import { Button } from 'react-bootstrap-4/lib'
+import Component from 'base-component'
+import Dropzone from 'react-dropzone'
+import Icon from 'icon'
+import React from 'react'
+import Upgrade from 'xoa-upgrade'
+import {
+  formatSize
+} from 'utils'
+import {
+  importConfig,
+  exportConfig
+} from 'xo'
+
+import styles from './index.css'
+
+// ===================================================================
+
+export default class Config extends Component {
+  componentWillMount () {
+    this.state = { importStatus: 'noFile' }
+
+    if (window.FileReader) {
+      this.reader = new window.FileReader()
+      this.reader.addEventListener('load', this._parseAndImport)
+      this.componentWillUnmount = () => this.reader.removeEventListener('load', this._parseAndImport)
+    }
+
+    exportConfig().then(config => {
+      this.setState({
+        url: `data:text/json;fileName=test.json;base64,${window.btoa(unescape(encodeURIComponent(JSON.stringify(config, null, 4))))}`
```

Comment:
It seems the length of this kind of URLs can be limited to 65KiB, which will much certainly cause issues.

We're going to change the server behavior to return a URL for the download.